src/math/quat.zig

```diff
@@ -15,9 +16,266 @@ pub fn Quat(comptime Scalar: type) type {
         /// The underlying Vec type, e.g. math.Vec4, math.Vec4h, math.Vec4d
         pub const Vec = vec.Vec(4, Scalar);
 
-        pub inline fn init(x: T, y: T, z: T, w: T) Quat(Scalar) {
+        /// The Vec type used to represent axes, e.g. math.Vec3
+        pub const Axis = vec.Vec(3, Scalar);
```

Comment:
I'd suggest we rename this to `Vec3` instead - although `Axis` is obviously correct, a Quaternion axis is after all just a vector and I think it'll be less confusing to less math-inclined folks to see it's a `Vec3` than `Axis`